### ISSUE TITLE
Add WM_ROLE to windows generated by flameshot

### ISSUE THREAD

**Issue by yggdr** [flameshot-org/flameshot#3635]
### Feature Description

Currently I cannot easily control which of the windows flameshot opens (like configuration, or the "Pin to desktop"-window) should be set to floating in my i3 configuration. With firefox e.g. I able to filter for `window_role="PictureInPicture"` to have the video player popup automatically in floating mode. In flameshot, I can only really filter for the entire `class='flameshot'`. 
Adding `WM_ROLE` to the various windows opened would make this selection more granular.

**Comment by mmahmoudian** [comment]
I agree that this can be further improved, but at the moment, at least for the windows I tested, it is possible to have granular window rules by using combination of window class and window title:

![bitmap](https://github.com/flameshot-org/flameshot/assets/390889/e41ddf3b-64d6-47b6-8e17-2c2badc47850)
